include/CObjectCache.h

```diff
@@ -9,58 +9,110 @@
 #include <functional>
 #include <map>
 #include <unordered_map>
-#include "IReferenceCounted.h"
 
 namespace irr { namespace core
 {
 
 namespace impl
 {
+    struct CMultiCache_tag {};
+
     template<template<typename...> class, template<typename...> class>
     struct is_same_templ : std::false_type {};
 
     template<template<typename...> class T>
     struct is_same_templ<T, T> : std::true_type {};
 
+    template<template<typename...> class T>
+    struct is_multi_container : std::false_type {};
+    template<>
+    struct is_multi_container<std::multimap> : std::true_type {};
+    template<>
+    struct is_multi_container<std::unordered_multimap> : std::true_type {};
+
+    template<template<typename...> class T>
+    struct is_assoc_container : std::false_type {};
+    template<>
+    struct is_assoc_container<std::map> : std::true_type {};
```

Comment:
Because it doesn't matter `CPreInsertionVerifier<>::verify()` for multicaches always returns `true` disregarding container type. But i'll add them just for logic sense.